pallets/asset-index/src/lib.rs

```diff
@@ -677,6 +717,43 @@ pub mod pallet {
 			share.checked_div(&nav).ok_or_else(|| ArithmeticError::Overflow.into())
 		}
 
+		/// Consolidate and clean deposits while completing withdraw
```

Comment:
Please add some more docs about the intention of this call, can probably use the description in the issue.